### PR TITLE
Correct subtest order

### PR DESCRIPTION
**Pull request by racheljay**
This PR fixes issue #3473 by correcting the order in which subtests appear. Previously they were appearing in alphabetical order. Now the subtest names are being imported from the row info on the table above, so they should appear in the same order.

Edit: Logic refactored to use event listeners instead of two-way data binding.

### CODE REVIEW

File: webapp/components/test-file-results.js [hunk 1]
```diff
@@ -82,7 +82,8 @@ class TestFileResults extends WPTFlags(LoadingState(PathInfo(
       },
       rows: {
         type: Array,
-        computed: 'computeRows(resultsTable, onlyShowDifferences)'
+        computed: 'computeRows(resultsTable, onlyShowDifferences)',
+        notify: true,
```
Comment (KyleJu) [inline webapp/components/test-file-results.js:86]: We are removing two-way data binding in Polymer, in preparation for migration to Lit. Instead, could you use addEventListener?  E.g. https://github.com/web-platform-tests/wpt.fyi/pull/3341/files

Comment (racheljay) [reply webapp/components/test-file-results.js:86]: I'll see if I can re-work some things to incorporate that 👍

Comment (racheljay) [reply webapp/components/test-file-results.js]: Event listeners have been added

File: webapp/views/wpt-results.js [hunk 2]
```diff
@@ -580,6 +585,7 @@ class WPTResults extends AmendMetadataMixin(Pluralizer(WPTColors(WPTFlags(PathIn
   connectedCallback() {
     super.connectedCallback();
     this.addEventListener('triagemetadata', this.reloadPendingMetadata);
+    window.addEventListener('getSubtestRows', this.handleGetSubtestRows.bind(this));
```
Comment (KyleJu) [inline webapp/views/wpt-results.js:588]:
```suggestion
    this.addEventListener('getSubtestRows', this.handleGetSubtestRows.bind(this));
```
We should bind it to the closest parent DOM

File: webapp/views/wpt-results.js [hunk 3]
```diff
@@ -1271,8 +1286,8 @@ class WPTResults extends AmendMetadataMixin(Pluralizer(WPTColors(WPTFlags(PathIn
 
   showNewHistoryClicked() {
     return () => {
-      this.showNewHistory = true;
-    };
+      this.showNewHistory = true
```
Comment (KyleJu) [inline webapp/views/wpt-results.js:1289]:
```suggestion
      this.showNewHistory = true;
```

File: webapp/components/new-test-results-history-grid.js [hunk 4]
```diff
@@ -80,7 +80,8 @@ class TestResultsGrid extends PathInfo(PolymerElement) {
       dataTable: Object,
       runIDs: Array,
       path: String,
-      showTestHistory: { type: Boolean, value: false }
+      showTestHistory: { type: Boolean, value: false },
```
Comment (KyleJu) [inline webapp/components/new-test-results-history-grid.js:83]: nit: formatting

File: webapp/components/new-test-results-history-grid.js [hunk 5]
```diff
@@ -166,45 +167,48 @@ class TestResultsGrid extends PathInfo(PolymerElement) {
...
-        const startDate = new Date(dataPoint.date);
-
-        // Use the next entry as the end date, or use present time if this
-        // is the last entry
-        let endDate = now;
-        if (i + 1 !== browserTestData[subtestName].length) {
-          const nextDataPoint = browserTestData[subtestName][i + 1];
-          endDate = new Date(nextDataPoint.date);
-        }
+    if (this.subtestNames) {
```
Comment (KyleJu) [inline webapp/components/new-test-results-history-grid.js:170]: Maybe we should check this in the beginning of the function `if (!browserTestData) || !this.subtestNames)`

File: webapp/components/test-file-results.js [hunk 6]
```diff
@@ -315,9 +315,18 @@ class TestFileResults extends WPTFlags(LoadingState(PathInfo(
     } else {
       this.subtestRowCount = 0;
     }
-
+    this._fireEvent('getSubtestRows', { rows });
```
Comment (KyleJu) [inline webapp/components/test-file-results.js:318]:
```suggestion
    this._fireEvent('subtestrows', { rows });
```
Should be a noun and all lowercase

File: webapp/views/wpt-results.js (hunk 2 above)
Comment (KyleJu) [reply webapp/views/wpt-results.js:588]: `getSubtestRows` should be changed according per comment above

Comment (racheljay) [reply webapp/views/wpt-results.js]: done 👍

File: webapp/views/wpt-results.js (hunk 3 above)
Comment (racheljay) [reply webapp/views/wpt-results.js]: done 👍

File: webapp/components/new-test-results-history-grid.js (hunk 4 above)
Comment (racheljay) [reply webapp/components/new-test-results-history-grid.js]: done 👍

File: webapp/components/test-file-results.js (hunk 6 above)
Comment (racheljay) [reply webapp/components/test-file-results.js]: done 👍

File: webapp/components/new-test-results-history-grid.js [hunk 7]
```diff
@@ -80,21 +80,25 @@ class TestResultsGrid extends PathInfo(PolymerElement) {
       dataTable: Object,
       runIDs: Array,
       path: String,
-      showTestHistory: { type: Boolean, value: false }
+      showTestHistory: {
+        type: Boolean,
+        value: false
```
Comment (KyleJu) [inline webapp/components/new-test-results-history-grid.js:85]:
```suggestion
        value: false,
```

**Review by KyleJu — APPROVED** [review]
LGMT with a nit! Great work!!

**Pull request merged**